tldw_Server_API/app/core/Chat/command_router.py

```
@@ -0,0 +1,329 @@
+    # Per-user per-command rate limiting
+    bucket = _acquire_bucket(ctx.user_id, cmd)
+    # Implement synchronous token check using underlying fields
+    now = time.time()
+    elapsed = max(0.0, now - getattr(bucket, "last_refill", now))
+    bucket.tokens = min(bucket.capacity, bucket.tokens + elapsed * bucket.refill_rate)
+    bucket.last_refill = now
+    if bucket.tokens >= 1:
+        bucket.tokens -= 1
+    else:
+        log_counter("chat_command_error", labels={"command": cmd, "reason": "rate_limited"})
+        try:
```

Ruff (0.14.4)

`try`-`except`-`pass` detected, consider logging the exception

(S110)